src/actor_critic_agent/knowledge.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+
+from torch.distributions.categorical import Categorical
+
+
+class ActorCritic(torch.nn.Module):
+    def __init__(self, num_inputs=12, num_outputs=4):
+        super(ActorCritic, self).__init__()
+
+        self.num_inputs = num_inputs
+        self.num_outputs = num_outputs
+
+        self.model_structure()
+        self.train()
+
+    def model_structure(self):
+        self.fc1 = nn.Linear(self.num_inputs, 120)
+        self.fc2 = nn.Linear(120, 60)
+        self.fc3 = nn.Linear(60, 30)
+
+        self.relu = nn.ReLU()
+
+        self.value = nn.Linear(30, 1)
+        self.policy = nn.Linear(30, self.num_outputs)
+
+    def forward(self, inputs):
+        x = inputs.view(-1, self.num_inputs)
+        x = self.relu(self.fc1(x))
+        x = self.relu(self.fc2(x))
+        x = self.relu(self.fc3(x))
+
+        return self.policy(x), self.value(x)
+
+
+class Knowledge():
+    def __init__(self, input_frames, action_space):
+        self.input_frames = input_frames
+        self.action_space = action_space
+        self.model = ActorCritic(self.input_frames, self.action_space)
+        self.model.double()
+        self.model = self.model.to('cpu')
+
+        self.gamma = 0.9
+        self.tau = 1.0
+        self.entropy_coef = 0.01
+        self.value_loss_coef = 0.5
+        self.learning_rate = 0.00001
```

Comment:
I think we should add a parameter for each variable instead of hard coded.

```suggestion
        self.gamma = param_gamma
        self.tau = param_tau
        self.entropy_coef = param_entropy_coef
        self.value_loss_coef = param_loss_coef
        self.learning_rate = param_lr
```

Reply:
This will be changed in the next code version.